dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -2665,6 +2829,24 @@ void setError(THD* thd, uint32_t errcode, string errmsg, gp_walk_info& gwi)
     clearStacks(gwi);
 }
 
+int setErrorAndReturn(gp_walk_info &gwi)
+{
+    // if this is dervied table process phase, mysql may have not developed the plan
+    // completely. Do not error and eventually mysql will call JOIN::exec() again.
+    // related to bug 2922. Need to find a way to skip calling rnd_init for derived table
+    // processing.
+    if (gwi.thd->derived_tables_processing)
+    {
+        // MCOL-2178 isUnion member only assigned, never used
+        //MIGR::infinidb_vtable.isUnion = false;
```

Comment:
Feel free to finally remove these two obsolete lines.

Reply:
Done.